### PR TITLE
Feature/2563 users guide html

### PR DESCRIPTION
**Pull request by bob-carpenter**
#### Submission Checklist

- [n/a] Run unit tests: `./runTests.py src/test/unit`
- [n/a] Run cpplint: `make cpplint`
- [x] Declare copyright holder and open-source license: see below

#### Summary

* Convert users guide from LaTeX/pdf to bookdown/HTML+pdf 
* Add 100 pages of intro to BDA with examples
* Change title from *Stan User's Guide* to *Bayesian Statistics using Stan* to reflect expanded scope

#### Intended Effect

* Provide HTML doc for searchability on the web
* Provide markdown-based doc for ease of updates
* Keep pdf form of doc

#### How to Verify

The build target 

```
> make doc
```

will make both versions, as well as the reference manual and function reference.  
It should be built during continuous integration as it is delegated to in `make manual`.

#### Side Effects

Future manual changes will need to go into bookdown, so anything in process will need to be updated.  

#### Documentation

Uh, yes.

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company):

Columbia University

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

**Comment by mitzimorris** [comment]
@seantalts - the CircleCi test is failing, so I updated the Dockerfile, but I don't understand how we've got the CircleCi box configured - it pulls in a docker image from seantalts/bookdown.   how does this image get updated?
could you please fix?  could you please explain how this works for future devs?

**Comment by mitzimorris** [comment]
I'm running a local docker with an updated docker file.  target `make users-guide` crashed with console message:

```
label: simple-ex-30 (with options) 
List of 3
 $ echo    : logi FALSE
 $ warnings: logi FALSE
 $ results : logi FALSE

Quitting from lines 1005-1020 (simple-examples.Rmd) 
Error in compileCode(f, code, language = language, verbose = verbose) : 
  Compilation ERROR, function(s)/method(s) not created! In file included from /usr/lib/R/site-library/BH/include/boost/config.hpp:39:0,
                 from /usr/lib/R/site-library/BH/include/boost/math/tools/config.hpp:13,
                 from /usr/lib/R/site-library/StanHeaders/include/stan/math/rev/core/var.hpp:7,
                 from /usr/lib/R/site-library/StanHeaders/include/stan/math/rev/core/gevv_vvv_vari.hpp:5,
                 from /usr/lib/R/site-library/StanHeaders/include/stan/math/rev/core.hpp:12,
                 from /usr/lib/R/site-library/StanHeaders/include/stan/math/rev/mat.hpp:4,
                 from /usr/lib/R/site-library/StanHeaders/include/stan/math.hpp:4,
                 from /usr/lib/R/site-library/StanHeaders/include/src/stan/model/model_header.hpp:4,
                 from file119f480357.cpp:8:
/usr/lib/R/site-library/BH/include/boost/config/compiler/gcc.hpp:186:0: warning: "BOOST_NO_CXX11_RVALUE_REFERENCES" redefined
 #  define BOOST_NO_CXX11_R
Calls: local ... stan_model -> cxxfunctionplus -> cxxfunction -> compileCode

Execution halted
Error in Rscript_render(f, render_args, render_meta) : 
  Failed to compile simple-examples.Rmd
Calls: <Anonymous> ... render_new_session -> tryCatch -> tryCatchList -> Rscript_render
```

upon re-running target `make users-guide` knitr was able to run chunk `simple-ex-30` and is now chugging along on chunk `simple-ex-33`.   seems like the problems are running out of memory and/or problems when trying to run on multiple cores.

File: .circleci/doc-docker/Dockerfile [hunk 1]
```diff
@@ -1,9 +1,17 @@
 FROM drmike/r-bookdown
 
+RUN Rscript -e 'install.packages("tufte")'
```
Comment (seantalts) [inline .circleci/doc-docker/Dockerfile]: Best practice to add these after the other install steps so that they can remain cached

**Comment by seantalts** [comment]
I think Travis just timed out (somewhat typical) and needs to be restarted. I have no idea why but I can't seem to find the restart job button from my account...

Updated the docker image on docker hub and will restart the circleci job. I created an issue to automate this process https://github.com/stan-dev/stan/issues/2641

**Comment by seantalts** [comment]
Yeah @mitzimorris I'm seeing it get "Killed" on CircleCI, I guess that is probably it running out of memory? I think it has 4GB though - how much are you seeing it need locally?

**Comment by seantalts** [comment]
I'm going to look into getting Jenkins set up building from the docker file directly, and hopefully we can assign a larger amount of RAM to the job.

**Comment by mitzimorris** [comment]
@seantalts - docker failing on Jenkins w/ message:

```

[Stan_PR-2640-SE3S4U5K4J3OD67XUS2KWABQNBI42IMUGHK4VJE5ADRHU4P4O46A] Running shell script
+ docker build -t seantalts/bookdown .circleci/doc-docker
Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
script returned exit code 1
```

**Comment by seantalts** [comment]
Okay, I got it building with docker thingies but it's getting this: 
http://d1m1s1b1.stat.columbia.edu:8080/blue/organizations/jenkins/Stan/detail/PR-2640/20/pipeline/#step-23-log-593

I think this is not a docker resource issue but am not 100% positive. Looks like it's some C++ error? Thoughts?

I will delete the circle CI info and move the dockerfile as well, though I'm not sure exactly where it should live.

**Comment by seantalts** [comment]
There is a discussion here of the current build issues on this branch: https://discourse.mc-stan.org/t/move-users-guide-to-new-repo/5733/

**Comment by bob-carpenter** [comment]
Moved the users guide to the `stan-book` repo.  Closing this.

**Pull request closed without merging**